gateleen-queue/src/main/java/org/swisspush/gateleen/queue/expiry/ExpiryCheckHandler.java

```diff
@@ -108,45 +127,43 @@ public static Integer getExpireAfter(MultiMap headers) {
      * @return queue-expire-after time in seconds or null if nothing is found
      */
     public static Integer getQueueExpireAfter(MultiMap headers) {
-        return getExpireValue(headers, QUEUE_EXPIRE_AFTER_HEADER);
+        try{
+            final Integer ans = getExpireValue( headers.get(QUEUE_EXPIRE_AFTER_HEADER) );
+            // Convert -1 to null
+            return (ans != null && ans == -1) ? null : ans;
+        }catch( NumberFormatException e ){
+            // Treat corrupt header same as it were not set at all.
+            // This is to keep backward compatibility to previous version of getExpireValue(MultiMap,String).
+            return null;
+        }
     }
 
     /**
-     * Extracts the value of the given header field.
-     * If the value can't be extracted (not found, invalid, and so on),
-     * null is returned.
-     *
-     * @param headers headers
-     * @param field the header field
-     * @return expire-after time in seconds or null if nothing is found
+     * @param expirationTimeout
+     *      Value as provided in the X-Expire-After header for example.
+     * @return
+     *      Expire-after time in seconds or null if header not present. This
+     *      possibly returns value -1 indicating "never expires".
+     * @throws NumberFormatException
+     *      In case the value contained in the header is not a valid
+     *      'X-Expire-After' value. A negative number or a non-numeric value
+     *      for example.
      */
-    private static Integer getExpireValue(MultiMap headers, String field) {
-        String expireHeaderValue = headers.get(field);
-
-        Integer value;
-
-        // no header set for the given field
-        if (expireHeaderValue == null) {
-            log.debug(field + " not defined");
-            value = null;
-        } else {
-            try {
-                value = Integer.parseInt(expireHeaderValue);
-
-                // redis returns an error if setex is called with negativ values
-                if (value < 0) {
-                    log.debug(field + " is a negative number: " + expireHeaderValue);
-                    value = null;
-                } else {
-                    log.debug("Setting " + field + " value to " + value + " seconds as defined in header " + field);
-                }
-            } catch (Exception e) {
-                log.warn(field + " is not a number: " + expireHeaderValue);
-                value = null;
+    private static Integer getExpireValue( final String expirationTimeout ) {
+        final Integer ans;
+        if( expirationTimeout == null ) {
+            // If we get null we'll answer with null.
+            ans = null;
+        }else{
+            // Value exists. Try parsing it (this will throw specified exception on failure).
+            ans = Integer.parseInt( expirationTimeout );
```

Comment:
Maybe rethink exception handling here. In the current implementation you make the exception handling not until _HookHandler_ class. No other util method in this class throws exceptions. We should stick to a single strategy.

Reply:
Same as https://github.com/swisspush/gateleen/pull/209#discussion_r178061152

But I'll change it :-)